jupyterhub/app.py

```diff
@@ -2123,6 +2127,12 @@ def _log_cls(name, cls):
         _log_cls("Authenticator", self.authenticator_class)
         _log_cls("Spawner", self.spawner_class)
 
+        self.event_log = EventLog(parent=self)
+
+        for schema_file in glob(os.path.join(here, 'event-schemas','*.json')):
```

Comment:
Should the schema source/definitions be a configurable trait?

Reply:
Great question!

The various extensions that can emit events from within JupyterHub (which is really what is hit by this code path), are:

1. JupyterHub Spawners
2. JupyterHub Authenticators
3. JupyterHub Proxies

All three will probably have their own schemas. They will have access to the same `event_log` object created here, and can register schemas based on where they have them stored. 

Is there a use case for registering schemas that no code is going to emit events with? If not, I think it's ok to leave that be not a traitlet. We can always expose that later if we have a use case.